src/test/java/com/booleanuk/core/BasketTest.java

```diff
@@ -2,7 +2,67 @@
 
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
 
-class BasketTest {
+class BasketTest
+{
+    private Basket basket;
+    private final ByteArrayOutputStream outputStreamCaptor = new ByteArrayOutputStream();
 
-}
+    @Test
+    public void addBagelToFullBasketCheck()
+    {
+        basket = new Basket(2);
+        String bagel1 = "apple";
+        String bagel2 = "orange";
+        String bagel3 = "tomato";
+
+        basket.addBagel(bagel1);
+        basket.addBagel(bagel2);
+        basket.addBagel(bagel3);
+
+        Assertions.assertFalse(basket.getBagels().contains(bagel3));
+    }
+
+
+    @Test
+    public void removeBagelCheck()
+    {
+        basket = new Basket(3);
+        String bagel1 = "apple";
+        String bagel2 = "orange";
+        String bagel3 = "tomato";
+        basket.addBagel(bagel1);
+        basket.addBagel(bagel2);
+
```

Comment:
This tests that removing a non-existent bagel has no effect on the existing list of bagels. You should also test that removing an existing bagel actually does so.